packages/cli-clean/src/clean.ts

```diff
@@ -0,0 +1,238 @@
+import {getLoader} from '@react-native-community/cli-tools';
+import type {Config as CLIConfig} from '@react-native-community/cli-types';
+import {spawn} from 'child_process';
+import {existsSync as fileExists, rmdir} from 'fs';
+import os from 'os';
+import path from 'path';
+import prompts from 'prompts';
+import {promisify} from 'util';
+
+type Args = {
+  include?: string;
+  projectRoot: string;
+  verifyCache?: boolean;
+};
+
+type Task = {
+  label: string;
+  action: () => Promise<void>;
+};
+
+type CLICommand = {
+  [key: string]: Task[];
+};
+
+const DEFAULT_CATEGORIES = ['metro', 'npm', 'watchman', 'yarn'];
+
+const rmdirAsync = promisify(rmdir);
+
+function cleanDir(directory: string): Promise<void> {
+  if (!fileExists(directory)) {
+    return Promise.resolve();
+  }
+
+  return rmdirAsync(directory, {maxRetries: 3, recursive: true});
+}
+
+function execute(command: string, args: string[], cwd: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const process = spawn(command, args, {
+      cwd,
+      stdio: ['inherit', null, null],
+    });
+
+    const stderr: Buffer[] = [];
+    process.stderr.on('data', (data) => {
+      stderr.push(data);
+    });
+
+    process.on('close', (code, signal) => {
+      if (code === 0) {
+        resolve();
+      } else if (stderr) {
+        reject(Buffer.concat(stderr).toString().trimEnd());
+      } else if (signal) {
+        reject(`Failed with signal ${signal}`);
+      } else {
+        reject(`Failed with exit code ${code}`);
+      }
+    });
+  });
+}
+
+function findPath(startPath: string, files: string[]): string | undefined {
+  // TODO: Find project files via `@react-native-community/cli`
+  for (const file of files) {
+    const filename = path.resolve(startPath, file);
+    if (fileExists(filename)) {
+      return filename;
+    }
+  }
+
+  return undefined;
+}
+
+async function promptForCaches(
+  commands: CLICommand,
+): Promise<string[] | undefined> {
+  const names: string[] = Object.keys(commands);
+  const {caches} = await prompts({
+    type: 'multiselect',
+    name: 'caches',
+    message: 'Select all caches to clean',
+    choices: names.map((cmd) => ({
+      title: cmd,
+      value: cmd,
+      selected: DEFAULT_CATEGORIES.includes(cmd),
+    })),
+    min: 1,
+  });
+  return caches;
+}
+
+export async function clean(
+  _argv: string[],
+  _config: CLIConfig,
+  cleanOptions: Args,
+): Promise<void> {
+  const {include, projectRoot, verifyCache} = cleanOptions;
+  if (!fileExists(projectRoot)) {
+    throw new Error(`Invalid path provided! ${projectRoot}`);
+  }
+
+  const npm = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
+  const yarn = os.platform() === 'win32' ? 'yarn.cmd' : 'yarn';
```

Comment:
In other packages, e.g. `cli-doctor` we use the `execa` package to handle this for us. Mind verifying it's ok and remove the platform checks? Hope this makes it unnecessary for the `execute` helper to exist

Reply:
Testing this on Windows now. Will reply once complete.

Reply:
Works on Windows. I also hid the CocoaPods option when on Windows and added hints:
```
% yarn react-native clean
? Select all caches to clean ›
Instructions:
    ↑/↓: Highlight option
    ←/→/[space]: Toggle selection
    a: Toggle all
    enter/return: Complete answer
◯   android (Android build caches, e.g. Gradle)
◯   cocoapods (CocoaPods cache)
◉   metro (Metro, haste-map caches)
◯   npm (`node_modules` folder in the current package, and optionally verify npm cache)
◉   watchman (Stop Watchman and delete its cache)
◯   yarn (Yarn cache)
```